src/audio/src.c

```diff
@@ -597,6 +597,8 @@ static struct comp_dev *src_new(struct sof_ipc_comp *comp)
 	cd->polyphase_func = src_polyphase_stage_cir;
 	src_polyphase_reset(&cd->src);
 
+	dev->output_rate = ipc_src->sink_rate;
```

Comment:
I did not see any changes about SOF_TKN_SRC_RATE_OUT in your topology part.
Could you give some simple workflow about how this is set, just take the using sof-apl-src-50khz-pcm512x.tplg as exmple.

Reply:
So the idea is, the component sets itself the output rate, if it _has_ fixed output rate. Like currently we have src and tone, that could use it. In other words it is not general feature that is set from topology for all components. As you noticed on other comment I rely on rzalloc to initialize it to 0 and components can set it after that (if they have fixed output rate).

Reply:
my point here is why mention 50khz in sof-apl-src-50khz-pcm512x.tplg.
I did not see any 50khz is src.
So you mean the actual output rate can comes from dai?
If it comes from dai, how could we do it only in downstream. The dai is the sink for src right?

Reply:
oh yes, I see what you mean. Currently you have to match the dai and src. so 1) dai doesn't specify its samplerate, you have to deduce it from the parameters. 2) you can't specify src output rate from top level topology, you should do that in pipe-src-playback, where it is currently hardcoded to 48kHz.

Reply:
so could you help me understand how sof-apl-src-50khz-pcm512x.tplg pass the 50kzh rate into the src part `dev->output_rate`

I think this is needed for the set_frame?